### PR TITLE
docs: restructure installation.rst + add OS X install instructions

### PR DESCRIPTION
**Pull request by yagebu**
I think the current `installation.rst` is too long and has too much duplicate information.

I've moved the dependencies to the sections where they are relevant and collected all the preparation steps (for 4 platforms now) as subsection under the new section "Installing the Dependencies".

Added install steps for OS X and added a link to the Arch Linux AUR package, apparently the first distribution package, yeah :)

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -4,153 +4,126 @@
...
-
-    # lz4 super fast compression support Headers + Library
     sudo dnf install lz4-devel
+    sudo dnf install fuse-devel fuse pkgconfig         # optional, for FUSE support
 
-    # optional: FUSE support - to mount backup archives
-    sudo dnf install fuse-devel fuse pkgconfig
-    
-    # optional: for unit testing
-    sudo dnf install fakeroot
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: why did you remove most of the korora dnf install commands?

File: docs/installation.rst [hunk 2]
```diff
@@ -159,41 +132,55 @@ You'll need at least (use the cygwin installer to fetch/install these):
...
 
 
 Installation (git)
 ------------------
 
+In addition to the dependencies installed above you will need Cython to compile
+the .pyx files to the respective .c files when using |project_name| code from
+git. If you want to run all tests, you also need fakeroot.  Both should be
+available through your package manager.
+
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: Cython from the linux dist's packages might be rather old, maybe it is better to use pip install cython.

File: docs/installation.rst (hunk 1 above)
Comment (yagebu) [inline docs/installation.rst]: I only removed the one for fakeroot. It's only needed for a dev installation and I put a note in the git installation instructions that fakeroot (and Cython) are required for that.

File: docs/installation.rst [hunk 3]
```diff
@@ -4,153 +4,126 @@
...
-simple use cases.
+Some Linux and BSD distributions might offer a ready-to-use ``borgbackup``
+package which can be installed with the package manager.  As |project_name| is
+still a young project, such a package might be not available for your system
+yet. Please ask package maintainers to build a package or, if you can package /
+submit it yourself, please help us with that!
 
-The llfuse_ python package is also required if you wish to mount an
-archive as a FUSE filesystem. Only FUSE >= 2.8.0 can support llfuse.
+* On **Arch Linux**, there is a package available in the AUR_.
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: maybe we do not want to maintain the list of linux packages here, that can change rather often.
dist users can easily check this in their package manager.

File: docs/installation.rst [hunk 4]
```diff
@@ -4,153 +4,126 @@
...
+The |project_name| binary is available on the releases_ page for the following
+platforms:
 
-Check https://github.com/borgbackup/borg/releases for available binaries.
+* **Linux**: glibc >= 2.13 (ok for most supported Linux releases)
+* **Mac OS X**: 10.10 (unknown whether it works for older releases)
+* **FreeBSD**: 10.2 (unknown whether it works for older releases)
 
+These binaries work without requiring specific installation steps. Just drop
+them into a folder on your ``PATH`` and then you can run ``borg``. If a new
```
Comment (ThomasWaldmann) [inline docs/installation.rst]: s/folder/directory/

File: docs/installation.rst (hunk 1 above)
Comment (ThomasWaldmann) [inline docs/installation.rst]: ah, sorry, it was just the comments that were removed. ok then.

File: docs/installation.rst (hunk 2 above)
Comment (yagebu) [inline docs/installation.rst]: Ok, will change this then.

File: docs/installation.rst (hunk 4 above)
Comment (yagebu) [inline docs/installation.rst]: Will do

File: docs/installation.rst (hunk 3 above)
Comment (yagebu) [inline docs/installation.rst]: The Arch AUR is not part of the official repos, so I think it's good to have this link somewhere. And at this stage users might not expect to find borg through their package manager and not even try.

Comment (ThomasWaldmann) [inline docs/installation.rst]: OK.

**Comment by anarcat** [comment]
great job - i wish we had unit tests for docs ... ;)

File: docs/installation.rst (hunk 2 above)
Comment (yagebu) [inline docs/installation.rst]: It's installed by the requirements file anyway

**Comment by yagebu** [comment]
Addressed comments.

Thanks for the review!

**Comment by ThomasWaldmann** [comment]
I am closing, rejecting this PR for now due to external reasons:

We messed up the gh repo and had to do a hard reset, so this PR is now messed up also.
See PR #290 about how to fix your PR. Contact @anarcat if you have questions.

YOUR changesets were fine, so please do a cleaned up PR with them again.
We are sorry for the troubles / additional work this caused, but there was no other way to get rid of some accident in master branch.
